File: Assets/Resources/_SFB_Particle Package 1/Scripts/demoParticleControl.js
﻿#pragma strict


var startPosition		: Vector3		= Vector3(0,0.1,0);		// Where should this particle start?
var startRotation		: Vector3		= Vector3(0,0,0);		// Rotation should this particle start?
var shootsTarget		: boolean		= false;				// If true, particle will shoot forward
var shootSpeed			: float			= 12.0;					// Multiplier of Time.deltaTime
var targetPosition		: Vector3		= Vector3(0,0.1,100);



function Start () {
	transform.position = startPosition;
}

function Update () {
	if (shootsTarget)
		transform.position = Vector3.MoveTowards(transform.position,targetPosition,shootSpeed * Time.deltaTime);
		//transform.position.z += shootSpeed * Time.deltaTime;
}

File: Assets/Resources/_SFB_Particle Package 1/Scripts/DemoParticleControl.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class DemoParticleControl : MonoBehaviour {

	public Vector3 startPosition = new Vector3(0f,0.1f,0f);		// Where should this particle start?
	public Vector3 startRotation = new Vector3(0f,0f,0f);		// Rotation should this particle start?
	public bool shootsTarget = false;				// If true, particle will shoot forward
	public float shootSpeed = 12.0f;					// Multiplier of Time.deltaTime
	public Vector3 targetPosition = new Vector3(0f,0.1f,100f);



	void Start () {
		transform.position = startPosition;
	}

	void Update () {
		if (shootsTarget) 
		{
			transform.position = Vector3.MoveTowards(transform.position,targetPosition,shootSpeed * Time.deltaTime);
		}
			
		//transform.position.z += shootSpeed * Time.deltaTime;
	}
}
